.circleci/circle-ci-config.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import copy
+import math
+import os
+import sys
+
+bin_dir = os.path.dirname(os.path.realpath(__file__))
+src_dir = bin_dir
+dep_dir = os.path.join(bin_dir, '.python3-deps')
+sys.path.append(dep_dir)
+
+try:
+    import yaml
+except ImportError:
+    import subprocess
+    # dependencies not installed; install them outside of the normal system
+
+    subprocess.check_call([sys.executable, '-m', 'pip', 'install', "--target=" + dep_dir, 'pyyaml'])
+
+    import yaml
```

Comment:
this is to make sure the script "just works", so if you don't have the package installed the script can move forward by installing local to this directory